evictiontest/workflow_cache_eviction_test.go

```diff
@@ -0,0 +1,178 @@
+	}
+	ctx = internal.WithActivityOptions(ctx, ao)
+	err := internal.ExecuteActivity(ctx, "testActivity").Get(ctx, nil)
+	if err != nil {
+		panic("Failed workflow")
+	}
+	return err
+}
+
+type (
+	CacheEvictionSuite struct {
+		suite.Suite
+		mockCtrl *gomock.Controller
+		service  *workflowservicetest.MockClient
+	}
+)
+
+// Test suite.
+func (s *CacheEvictionSuite) SetupTest() {
+	s.mockCtrl = gomock.NewController(s.T())
+	s.service = workflowservicetest.NewMockClient(s.mockCtrl)
+}
+
+func (s *CacheEvictionSuite) TearDownTest() {
+	s.mockCtrl.Finish() // assert mock’s expectations
+}
+
+func TestWorkersTestSuite(t *testing.T) {
+	formatter := &log.TextFormatter{}
+	formatter.FullTimestamp = true
+	log.SetFormatter(formatter)
+	log.SetLevel(log.DebugLevel)
+	suite.Run(t, new(CacheEvictionSuite))
+}
+
+// this is the mock for yarpcCallOptions, make sure length are the same
+var callOptions = []interface{}{gomock.Any(), gomock.Any(), gomock.Any()}
+
+func createTestEventWorkflowExecutionStarted(eventID int64, attr *m.WorkflowExecutionStartedEventAttributes) *m.HistoryEvent {
+	return &m.HistoryEvent{EventId: common.Int64Ptr(eventID), EventType: common.EventTypePtr(m.EventTypeWorkflowExecutionStarted), WorkflowExecutionStartedEventAttributes: attr}
+}
+
+func createTestEventDecisionTaskScheduled(eventID int64, attr *m.DecisionTaskScheduledEventAttributes) *m.HistoryEvent {
+	return &m.HistoryEvent{
+		EventId:   common.Int64Ptr(eventID),
+		EventType: common.EventTypePtr(m.EventTypeDecisionTaskScheduled),
+		DecisionTaskScheduledEventAttributes: attr}
+}
+
+func (s *CacheEvictionSuite) TestResetStickyOnEviction() {
+	testEvents := []*m.HistoryEvent{
+		createTestEventWorkflowExecutionStarted(1, &m.WorkflowExecutionStartedEventAttributes{
+			TaskList: &m.TaskList{Name: common.StringPtr("tasklist")},
+		}),
+		createTestEventDecisionTaskScheduled(2, &m.DecisionTaskScheduledEventAttributes{}),
+	}
+
+	var taskCounter atomic.Int32 // lambda variable to keep count
+	// mock that manufactures unique decision tasks
+	mockPollForDecisionTask := func(ctx context.Context, _PollRequest *m.PollForDecisionTaskRequest, opts ...yarpc.CallOption,
+	) (success *m.PollForDecisionTaskResponse, err error) {
+		taskID := taskCounter.Inc()
+		workflowID := common.StringPtr("testID" + strconv.Itoa(int(taskID)))
+		runID := common.StringPtr("runID" + strconv.Itoa(int(taskID)))
+		// how we initialize the response here is the result of a series of trial and error
+		// the goal is we want to fabricate a response that looks real enough to our worker
+		// that it will actually go along with processing it instead of just tossing it out
+		// after polling it or giving an error
+		ret := &m.PollForDecisionTaskResponse{
+			TaskToken:              make([]byte, 5),
+			WorkflowExecution:      &m.WorkflowExecution{WorkflowId: workflowID, RunId: runID},
+			WorkflowType:           &m.WorkflowType{Name: common.StringPtr("go.uber.org/cadence/internal.testReplayWorkflow")},
```

Comment:
the name should include package evictiontest? Another way is to register workflow with custom name so you don't have to use full name.

Reply:
So actually in this instance it doesn't matter because workflow lookup happens after creating & inserting workflow cache and apparently we don't evict when workflow execution fails due to this.

Reply:
please file an issue to get this fixed.